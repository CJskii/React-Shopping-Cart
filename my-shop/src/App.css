@tailwind base;
@tailwind components;
@tailwind utilities;

* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.App {
  background-image: url("../public/images/background.svg");
  min-height: 100vh;
}
