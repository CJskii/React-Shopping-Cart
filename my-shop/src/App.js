import "./App.css";
import Navbar from "./Components/Navbar";
import { Routes, Route } from "react-router-dom";
import Home from "./Components/Home/Home";
import Shop from "./Components/Products/Shop";
import ContactUs from "./Components/ContactUs";
import Footer from "./Components/Footer";
import CategoryItems from "./Components/Products/Category/CategoryItems";
import ItemDetails from "./Components/Products/Category/ItemDetails";

function App() {
  return (
    <div
      data-testid="app"
      className="App min-w-full min-h-full flex flex-col items-center justify-center "
    >
      <Navbar />
      <Routes>
        <Route exact path="/" element={<Home />} />
        <Route exact path="/products" element={<Shop />} />
        <Route exact path="/contact" element={<ContactUs />} />
        <Route path="/products/:name" element={<CategoryItems />} />
        <Route path="/products/:name/:id" element={<ItemDetails />} />
      </Routes>
      <Footer />
    </div>
  );
}

export default App;
