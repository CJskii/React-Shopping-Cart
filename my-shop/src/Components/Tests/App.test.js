import { getByTestId, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../../App";
import { MemoryRouter } from "react-router-dom";

describe("Render main components", () => {
  it("should render app", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    const app = getByTestId("app");
    expect(app).toBeInTheDocument();
  });

  it("should render navbar", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    const navbarElement = getByTestId("navbar");
    expect(navbarElement).toBeInTheDocument();
  });

  it("should render footer", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    const footer = getByTestId("footer");
    expect(footer).toBeInTheDocument();
  });
});
