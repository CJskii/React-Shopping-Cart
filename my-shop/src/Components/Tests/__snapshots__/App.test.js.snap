// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render main components should be always the same 1`] = `undefined`;
